modules/userId/index.js

```diff
@@ -481,15 +482,40 @@ function getCombinedSubmoduleIdsForBidder(submodules, bidder) {
   if (!Array.isArray(submodules) || !submodules.length || !bidder) {
     return {};
   }
-  return submodules
+  const eligibleSubmodules = submodules
     .filter(i => !i.config.bidders || !isArray(i.config.bidders) || includes(i.config.bidders, bidder))
+
+  return getPrioritizedCombinedSubmoduleIds(eligibleSubmodules);
+}
+
+/**
+ * @param {SubmoduleContainer[]} submodules
+ */
+function getPrioritizedCombinedSubmoduleIds(submodules) {
+  const combinedIdStates = submodules
     .filter(i => isPlainObject(i.idObj) && Object.keys(i.idObj).length)
     .reduce((carry, i) => {
       Object.keys(i.idObj).forEach(key => {
-        carry[key] = i.idObj[key];
+        const currentIdPriorityByName = idPriority[key]?.reverse().indexOf(i.submodule.name);
```

Comment:
actually, doing the reverse only once may be a requirement and not just a suggestion - because IIRC it modifies the array in place so on the second call you'd actually be restoring it to the original order.